lib/sync.js

```diff
@@ -543,7 +639,23 @@ SyncApi.prototype._processRoomEvents = function(room, stateEventList,
     // execute the timeline events, this will begin to diverge the current state
     // if the timeline has any state events in it.
     room.addEventsToTimeline(timelineEventList);
+};
 
+/**
+ * @return {string}
+ */
+SyncApi.prototype._getGuestFilter = function() {
+    var guestRooms = this.client._guestRooms; // FIXME: horrible gut-wrenching
+    if (!guestRooms) {
+        return "{}";
+    }
+    return JSON.stringify({
+        room: {
+            timeline: {
+                limit: 20
+            }
+        }
+    });
```

Comment:
If I've read this correctly, we add a limit of 20 to the sync query if we're peeking into a room (which is done via a separate query?) Why?